### PR TITLE
fix: robust eigendecomposition and bound-aware sensitivity for grad-simp

### PR DESCRIPTION
**Pull request by ericchansen**
## Summary

Robust eigendecomposition fix, lazy JAX imports to prevent CUDA abuse in tests, and a comprehensive docs overhaul that removes all fabricated claims and grounds every assertion in evidence.

### Robust eigendecomposition (`q2mm/models/hessian.py`)
- Symmetrise Hessian before eigendecomposition: `0.5*(H+H.T)`
- Detect non-finite values (NaN/Inf) in Hessian
- New `on_error` parameter: `"raise"` (default, preserves existing behavior) or `"penalty"` (returns 1e5 cm⁻¹ penalty frequencies)

### Bound-aware sensitivity (`q2mm/optimizers/cycling.py`)
- Symmetric step shrinking: `h_eff = min(step, upper-x0, x0-lower)` preserves O(h²) central-difference accuracy
- Parameters at bounds get `simp_var=inf`, ranking them last and effectively excluding them from simplex selection
- `OptimizationLoop.run()` automatically enables penalty mode during optimization

### Prevent CUDA initialization in tests (`test/conftest.py`)
- `JAX_PLATFORMS=cpu` set in `pytest_configure` as safety net
- Test files use `find_spec` for skip conditions instead of importing JAX at module scope
- Lazy JAX imports throughout `_jax_common.py`, `jax_engine.py`, `jax_md_engine.py`, `batched.py`

### Caller threading
- `on_error` parameter threaded through `MMEngine.frequencies()` → `FrequencyEvaluator.compute()` → `ObjectiveFunction` for all backends (OpenMM, Tinker, JAX, batched)

## Documentation overhaul

Every claim is now grounded in evidence -- committed benchmark JSONs, published papers, or source code links. Fabricated numbers and unverifiable assertions have been removed.

### New pages
- **Theory & Methods**: full pipeline (Seminario, eigenvalue methods A-E, objective function, optimization, numerical safety) with SVG figures (pure vector, light/dark variants)
- **Architecture**: data models, canonical units, pluggable backends
- **Optimization guide**: single-shot, grad-simp cycling, manual subspace
- **Q2MM v5 Improvements**: comparison table with evidence links for every claim
- **Platform support**: compatibility matrix, GPU setup, common issues
- **Published FF validation**: Check 1/Check 2 overview written for grad-student audience
- **Tutorial**: full workflow from QM data to optimised force field
- **Psi4 backend page**

### Fixes to existing pages
- Removed fabricated "37x parameter quality" numbers from Rh-enamide benchmarks (ephemeral computation, never committed)
- Removed false claims: "no GPU support" (unverifiable), "no TS eigenvalue methods" (wrong -- upstream supports C/D/E), "no diagnostics" (upstream has `tools/` with 15 scripts)
- Softened unverifiable claims: "No analytical gradients" → "Not evident"
- Standardised all h2/h3 headers to sentence case (75 headers across 17 files)
- Added Hansen sulfamide paper to references (now 18 entries)
- Fixed QFUERZA year to 2025

### Nav restructure
- Moved comparison and platform support pages to top-level nav
- Moved file locations to match (`docs/comparison.md`, `docs/platform-support.md`)

## Infrastructure

- Consolidated `benchmark_results/` into `benchmarks/` (one canonical location per AGENTS.md)
- Benchmark data tracked in git (not gitignored)
- Added AGENTS.md rules: alpha discipline, evidence-grounded claims (§2.8), comparison substantiation (§2.9), grad-student audience (§2.10), publication metadata validation (§10)
- Gitignore `uv.lock` and agent session artifacts
- Renamed `mkdocs.yml` → `properdocs.yml`

## Testing

- 14 new unit tests in `test/test_robust_eigendecomp.py`
- Full test suite: **742 passed, 0 failed**
- Lint + format: clean

### CODE REVIEW

File: q2mm/optimizers/cycling.py [hunk 1]
```diff
@@ -294,20 +303,44 @@ def compute_sensitivity(
...
-    # Build the full (2N+1) parameter matrix: [x0, x0+h0, x0-h0, x0+h1, x0-h1, ...]
+    # Compute effective step sizes: shrink to stay within bounds
+    eff_steps = step_sizes.copy()
+    if bounds is not None:
+        lower = np.array([b[0] for b in bounds])
+        upper = np.array([b[1] for b in bounds])
+        # Largest symmetric step that keeps x0 ± h within [lower, upper]
+        max_up = upper - x0
+        max_down = x0 - lower
+        max_symmetric = np.minimum(max_up, max_down)
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/optimizers/cycling.py]: Fixed in 2440e70. Added length check and lower<=upper validation.

File: test/test_robust_eigendecomp.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        obj = MagicMock()
+        obj.forcefield = ff
+
+        # Quadratic objective: f(x) = sum((x - x0)^2)
+        x0 = ff.get_param_vector()
+
+        def mock_call(pvec: np.ndarray) -> float:
+            return float(np.sum((np.asarray(pvec) - x0) ** 2))
+
+        obj.__call__ = mock_call
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply test/test_robust_eigendecomp.py]: Fixed in 2440e70. Changed to side_effect as suggested.

File: test/conftest.py [hunk 3]
```diff
@@ -108,6 +104,17 @@ def pytest_configure(config: pytest.Config) -> None:
...
+    # Prevent JAX from allocating GPU memory when JAX tests are excluded.
+    # JAX initialises CUDA at import time, consuming ~25 GiB of VRAM even
+    # when no JAX tests will run.  Setting JAX_PLATFORMS=cpu before import
+    # keeps the GPU free for other work.
+    import os
+
+    markexpr = getattr(config.option, "markexpr", "") or ""
+    jax_excluded = "not" in markexpr and ("jax" in markexpr)
+    if jax_excluded and "JAX_PLATFORMS" not in os.environ:
+        os.environ["JAX_PLATFORMS"] = "cpu"
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply test/conftest.py]: Fixed in 2440e70. Now uses regex with word boundaries to match "not jax" or "not (...jax...)" without false-positiving on "not slow and jax".
